include/neighbor.h

```diff
@@ -13,57 +13,100 @@ namespace diskann {
   struct Neighbor {
     unsigned id;
     float    distance;
-    bool     flag;
+    bool     checked;
 
     Neighbor() = default;
```

Comment:
This default constructor isn't initializing variables in the class. I was seeing crashes during index build because cur_ was set to a random value.